src/ruby/puppet-server-lib/puppet/server/config.rb

```diff
@@ -25,6 +25,15 @@ def self.initialize_puppet_server(puppet_server_config)
       Puppet::Util::Profiler.add_profiler(@profiler)
     end
 
+    if puppet_server_config["shared_terminus_state"]
+      begin
+        require 'puppet/util/puppetdb/shared_terminus_state'
+        Puppet::Util::Puppetdb::SharedTerminusState.shared_terminus_state = puppet_server_config["shared_terminus_state"]
```

Comment:
@camlow325  that's a good thought, and a good thing to keep in mind, but yeah, I don't think we can answer that question now, and it shouldn't be too big of a deal to move it around later.  At this point I don't think it's worth holding up @mullr any longer for that particular concern.